pynamodb/attributes.py

```diff
@@ -257,3 +257,128 @@ def deserialize(self, value):
         Takes a UTC datetime string and returns a datetime object
         """
         return parse(value, dayfirst=False).datetime
+
+
+class MapAttribute(Attribute):
+
+    attr_type = MAP
+    model = None
+
+    def __init__(self,
+                 hash_key=False,
+                 range_key=False,
+                 null=None,
+                 default=None,
+                 attr_name=None,
+                 model=None):
+        super(MapAttribute, self).__init__(hash_key=hash_key,
+                                           range_key=range_key,
+                                           null=null,
+                                           default=default,
+                                           attr_name=attr_name)
+        if model:
+            self.model = model
+
+    def serialize(self, values):
+        """
+        Encode the given list of numbers into a list of AttributeValue types.
+        """
+        rval = dict()
+        for k in values:
+            v = values[k]
+            if v is None:
+                rval[k] = {'NULL': bool(1)}
+            elif type(v) is dict:
+                rval[k] = {'M': MapAttribute().serialize(v)}
+            elif type(v) is list:
+                rval[k] = {'L': ListAttribute().serialize(v)}
+            elif type(v) is float or type(v) is int or type(v) is long:
+                rval[k] = {'N': NumberAttribute().serialize(v)}
+            elif type(v) is str or type(v) is unicode or type(v) is basestring:
+                rval[k] = {'S': UnicodeAttribute().serialize(v)}
+            elif type(v) is bool:
+                value_to_dump = bool(0)
+                if v:
+                    value_to_dump = bool(1)
+
+                rval[k] = {'BOOL': value_to_dump}
+            else:
+                raise Exception('Unknown value: ' + str(v))
+
+        return rval
+
+    def deserialize(self, values):
```

Comment:
We need to check None here. When reading from records written from other sources a None may be passed in here.